app/controllers/api/v1/targets_controller.rb

```diff
@@ -0,0 +1,16 @@
+module Api
+  module V1
+    class TargetsController < Api::V1::ApiController
+      def create
+        authorize Target
+        @target = Target.create!(target_params)
+      end
+
+      private
+
+      def target_params
+        params.require(:target).permit(:title, :radius, :lat, :lon, :topic_id, :user_id)
```

Comment:
I'm not sure, but since you are authorizing it with the `current_user` this is not necessary and you can infer it?

```suggestion
        params.require(:target).permit(:title, :radius, :lat, :lon, :topic_id)
```

Reply:
I´m getting a "user must exist" bad request error if I remove it.

Reply:
On create method you can do `current_user.targets.create!(targets_params)`.

Reply:
Done it!